mvp_onboarding.client/src/components/CustomerTable.jsx

```diff
@@ -0,0 +1,354 @@
+import React, { useEffect, useState } from "react";
+import { getCustomers, deleteCustomer, createCustomer, updateCustomer } from "../services/customerService";
+import {
+  Table,
+  Button,
+  Icon,
+  Modal,
+  Form,
+  Input,
+  Pagination,
+  Dropdown
+} from "semantic-ui-react";
+
+const CustomerTable = () => {
+  const [customers, setCustomers] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [deleteOpen, setDeleteOpen] = useState(false);
+  const [newOpen, setNewOpen] = useState(false);
+  const [editOpen, setEditOpen] = useState(false);
+  const [selectedCustomer, setSelectedCustomer] = useState(null);
+  const [createIsDisabled, setCreateIsDisabled] = useState(true);
+  const [editIsDisabled, setEditIsDisabled] = useState(true);
+  const [selectedAddress, setSelectedAddress] = useState("");
+  const [selectedName, setSelectedName] = useState("");
+
+  const [sortColumn, setSortColumn] = useState('Name');
+  const [sortDirection, setSortDirection] = useState('asc');
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalCount, setTotalCount] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+  const pageSizeOptions = [
+    { key: 10, text: '10', value: 10 },
+    { key: 20, text: '20', value: 20 },
+    { key: 50, text: '50', value: 50 },
+  ];
+
+  const handleSort = (column, direction) => {
+    setSortColumn(column);
+    setSortDirection(direction);
+  }
+
+  useEffect(() => {
+    fetchCustomers(currentPage, pageSize, sortColumn, sortDirection);
+  }, [currentPage, pageSize, sortColumn, sortDirection]);
+
+  useEffect(() => {
+    setCreateIsDisabled(invalidSelectedCustomer);
+    setEditIsDisabled(invalidSelectedCustomer);
+  }, [selectedAddress, selectedName]);
```

Comment:
I suggest to add the hooks below before the return method. Otherwise, this might cause some rendering issues. For example, you might encounter some errors like: **fetchCustomer** is not a function or rendering fewer hooks.